bee-message/src/output/foundry.rs

```diff
@@ -383,17 +397,73 @@ impl StateTransitionVerifier for FoundryOutput {
             return Err(StateTransitionError::MutatedImmutableField);
         }
 
+        let token_id = next_state.token_id();
+        let input_tokens = context.input_native_tokens.get(&token_id).copied().unwrap_or_default();
+        let output_tokens = context.output_native_tokens.get(&token_id).copied().unwrap_or_default();
         let TokenScheme::Simple(ref current_token_scheme) = current_state.token_scheme;
         let TokenScheme::Simple(ref next_token_scheme) = next_state.token_scheme;
 
-        if current_token_scheme.maximum_supply() != next_token_scheme.maximum_supply() {
+        if current_token_scheme.minted_tokens() > next_token_scheme.minted_tokens()
+            || current_token_scheme.melted_tokens() > next_token_scheme.melted_tokens()
+            || current_token_scheme.maximum_supply() != next_token_scheme.maximum_supply()
+        {
             return Err(StateTransitionError::MutatedImmutableField);
         }
 
+        match input_tokens.cmp(&output_tokens) {
+            Ordering::Less => {
+                // Minting
+
+                if &(current_token_scheme.minted_tokens() + (output_tokens - input_tokens))
+                    != next_token_scheme.minted_tokens()
+                    || current_token_scheme.melted_tokens() != next_token_scheme.melted_tokens()
+                {
+                    return Err(StateTransitionError::InconsistentNativeTokens);
```

Comment:
Why do all three different state transitions create the same error? Wouldn't it make sense to have:
```rust
StateTransitionError::InconsistentMinting;
StateTransitionError::InconsistentTransitioning;
StateTransitionError::InconsistentMelting;
```
or something similar to make it easier to pinpoint what went wrong?

Reply:
Done